### PR TITLE
Update configs for tokamax

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

* clean up TODO in attention, as we have bump it up to 0.8.0 for github runners
* move tokamax ragged_dot to a different branch, so:

1) with flags `use_tokamax_gmm=True, sparse_matmul=True` --> tokamax ragged_dot
2) with flags `use_tokamax_gmm=False, sparse_matmul=True, megablox=True` --> original megablox
3) with flags `use_tokamax_gmm=False, sparse_matmul=True, megablox=False` --> JAX ragged_dot

# Tests

Expect existing tests are passing 

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by khatwanimohit — APPROVED** [review]

File: src/MaxText/layers/moe.py [hunk 1]
```diff
@@ -809,17 +809,17 @@ def gmm(inputs, kernel, tiling, group_sizes, expert_assignments):
...
-              lhs=inputs,
-              rhs=kernel,
-              group_sizes=group_sizes,
-              precision=jax.lax.Precision.DEFAULT,
-              preferred_element_type=self.dtype,
-              implementation="mosaic",
-          )
-        else:
+      if self.config.use_tokamax_gmm:
+        output = tokamax_api.ragged_dot(  #  pylint: disable=possibly-used-before-assignment
```
Comment (khatwanimohit) [inline src/MaxText/layers/moe.py]: can you also remove this pylint disable check and take out the conditional import of tokamax_api on top of this file. This conditions were only written temporarily until maxtext upgraded to jax=0.8.0

Comment (RissyRan) [reply src/MaxText/layers/moe.py]: Sounds good. Will update.
